### DISCUSSION TITLE
Want to setup locally ejabbered codebase [os ubuntu 20.04]

### DISCUSSION THREAD

**Discussion by zeeshanrafiqrana** [processone/ejabberd#4099] [Questions] [answered]
Hello, Ejabberd Community!

I'm a software engineer from Pakistan and an enthusiastic learner about XMPP software. I'd like to set up the Ejabberd codebase locally to learn and contribute to your community.

Could you please provide me with any documentation that explains how to easily set up the Ejabberd server locally with a PostgreSQL database? Your assistance is greatly appreciated. Thank you!

**Comment by badlop** [comment, marked as answer]
If you plan to modify the source code, you can follow the [compilation instructions in the ejabberd Docs](https://docs.ejabberd.im/admin/installation/#source-code).

If you only plan to use it without modifying the source code, you can install ejabberd using a binary installer, a container, or a package from your operating system, see the alternatives in [Installation](https://docs.ejabberd.im/admin/installation/).
